compositor_common/src/error.rs

```diff
@@ -44,6 +44,58 @@ impl Display for UnusedNodesError {
     }
 }
 
+#[derive(Debug, thiserror::Error, PartialEq, Eq)]
+pub enum ConstraintsValidationError {
+    #[error("Specification of node \"{1}\" doesn't satisfy node input pads count constraint.")]
```

Comment:
I would merge InputsCountValidationError into ConstraintsValidationError. If you prefer to keep them separate then make it transparent.

I would like the user to see sth like this:
- Specification of node \"{1}\" is invalid / Specification of node \"{1}\" failed constraint validation.
- Node does not expect any input pads. / Node does not expect any input pads, but some were provided.